application/src/main/java/org/togetherjava/tjbot/features/moderation/TransferQuestionCommand.java

```diff
@@ -0,0 +1,220 @@
+
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.features.BotCommandAdapter;
+import org.togetherjava.tjbot.features.CommandVisibility;
+import org.togetherjava.tjbot.features.MessageContextCommand;
+import org.togetherjava.tjbot.features.utils.StringDistances;
+
+import java.awt.Color;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.regex.Pattern;
+
+public final class TransferQuestionCommand extends BotCommandAdapter
+        implements MessageContextCommand {
+    private static final Logger logger = LoggerFactory.getLogger(TransferQuestionCommand.class);
+    private static final String COMMAND_NAME = "transfer-question";
+    private static final String TRANSFER_QUESTION_TITLE_ID = "transferID";
+    private static final String TRANSFER_QUESTION_INPUT_ID = "transferQuestion";
+    private static final String TRANSFER_QUESTION_TAG = "tags";
+    private static final int TITLE_MAX_LENGTH = 50;
+    private static final Pattern TITLE_COMPACT_REMOVAL_PATTERN = Pattern.compile("\\W");
+    private static final int TITLE_COMPACT_LENGTH_MIN = 2;
+    private static final int TITLE_COMPACT_LENGTH_MAX = 30;
+    private final Predicate<String> isHelpForumName;
+    private final List<String> defaultTags;
+    private static final Color EMBED_COLOR = new Color(50, 164, 168);
+
+
+    /**
+     * Creates a new instance.
+     */
+    public TransferQuestionCommand(Config config) {
+        super(Commands.message(COMMAND_NAME), CommandVisibility.GUILD);
+
+        isHelpForumName =
+                Pattern.compile(config.getHelpSystem().getHelpForumPattern()).asMatchPredicate();
+
+        defaultTags = config.getHelpSystem().getCategories();
+    }
+
+    @Override
+    public void onMessageContext(MessageContextInteractionEvent event) {
+        String originalMessage = event.getTarget().getContentRaw();
+        String originalMessageId = event.getTarget().getId();
+        String originalChannelId = event.getChannel().getId();
+        String authorId = event.getTarget().getAuthor().getId();
+
+        TextInput transferQuestionTitle =
+                TextInput.create(TRANSFER_QUESTION_TITLE_ID, "Title", TextInputStyle.SHORT)
+                    .setMaxLength(70)
+                    .setMinLength(8)
+                    .setValue(createTitle(originalMessage))
+                    .build();
+
+        TextInput transferQuestionInput = TextInput
+            .create(TRANSFER_QUESTION_INPUT_ID, "Transfer question menu", TextInputStyle.PARAGRAPH)
+            .setValue(originalMessage)
+            .setRequiredRange(3, 2000)
+            .build();
+
+        TextInput transferQuestionTag = TextInput
+            .create(TRANSFER_QUESTION_TAG, "Transfer question tags", TextInputStyle.SHORT)
+            .setValue("Java")
+            .build();
+
+        String transferQuestionModalComponentID =
```

Comment:
tip on naming. the context that this is about transferring questions is clear based on the class name already. everyone reading this class knows that we are now transffering a question.

hence, its obsolete and rather distracting to prefix all ur variables with `transferQuestion` (applies to the other variables here as well)

Reply:
Should i at least keep "modal" as prefix? otherwise the variables would be "title", "input", "tag".

Reply:
u can keep it. it just depends how clear the context is. suppose u have the following:
```java
void createModal() {
  String title = "...";
  ...
}
```
in this case its absolutely clear that the title is the title of the modal, bc the method is literally giving the context. in this case, `modalTitle` would be kinda unecessary.

but it doesnt hurt that much, since its so much shorter. im okay with that